stdlib/public/core/AnyHashable.swift

```diff
@@ -107,22 +107,31 @@ internal struct _ConcreteHashableBox<Base: Hashable>: _AnyHashableBox {
 /// A type-erased hashable value.
 ///
 /// The `AnyHashable` type forwards equality comparisons and hashing operations
-/// to an underlying hashable value, hiding its specific underlying type.
+/// to an underlying hashable value, hiding the type of the wrapped value.
+///
+/// For types that support conversions between each other using `as` or `as?`
+/// (such as `Int` and `NSNumber`), `AnyHashable` treats their values
+/// equivalently when type-erased by forwarding operations to canonical
+/// representations of the wrapped values.
```

Comment:
I like this edit better.  Some comments and a possible rewrite:

> `AnyHashable` normalizes their values

Whose values?  I think we mean the value that was wrapped in `AnyHashable`?

```
/// When `as` conversion is possible between two types (such as from `Int` to
/// `NSNumber` or between `String` and `NSString`), `AnyHashable` normalizes 
/// the wrapped value into a canonical representation, to ensure that instances of either
/// type that represent the same value  compare as equal.  For example, 
/// `AnyHashable(Int(5))` is equal to  `AnyHashable(NSNumber(5))`.
```

Reply:
I'm wary of using the term "normalize" here, particularly in the context of strings, where it could be misinterpreted as implying Unicode normalization. More generally, the statement that `AnyHashable` normalizes the wrapped value means, at face value, that the underlying wrapped `base` value is permanently changed--which it is not. `AnyHashable` merely __uses__ another ("canonical") value for the purposes of its comparison and hashing operations, but `base` is unchanged:

```swift

import Foundation

let x = "café" as NSString
let y = "cafe\u{301}" as NSString
x == y // false
AnyHashable(x) == AnyHashable(y) // true
AnyHashable(x).base as! NSString == AnyHashable(y).base as! NSString // false
```

I have adapted the wording accordingly so as to avoid giving this impression; I have incorporated the remainder of the suggested changes.